### DISCUSSION TITLE
Additional Documentation to Set Up <extern> on Windows 10 with MATLAB

### DISCUSSION THREAD

**Discussion by corsairguy275** [cyberbotics/webots#5118] [Q&A]
In the Webots documentation "Running Extern Robot Controllers" there are two tables provided to set up the Environment Variables for Windows and MATLAB.  For a newbie, such as myself, is there additional information on how and where to set these variables?  A step-by-step video in Youtube on your channel would be very helpful for this.

My O/S is Windows 10, MATLAB R2022a and Webots R2022a.  Thanks in advance for your help.

**Comment by omichel** [comment]
There is no such video tutorial, but we may help you here. Setting environment variables on Windows can be done from the Windows Settings, there are many tutorials online explaining how to do this, see for example [this one](https://www.alphr.com/environment-variables-windows-10/).

**Reply by corsairguy275** [reply]
Thank you for your reply.  The link was helpful for me to understand the Windows Environment Variables.  If I may ask another question, do we set the Environment Variables through the Windows System Settings, as the article shows, or do you do it through the "setenv( )" command in MATLAB?

**Reply by omichel** [reply]
Both work equally. It is maybe cleaner to set it inside your MATLAB program, so that it doesn't pollute the global environment and will remain local to your MATLAB program.

**Reply by corsairguy275** [reply]
Thank you again for your help.  If I may ask an additional question.  I've set up the Environment Variables--I think correctly.  When I run the "launcher.m" script in MATLAB, I get the following error  ( **** characters inserted for privacy):

>> getenv('WEBOTS_PROJECT')
ans =

    'C:\Users\*****\*****\MyProjects\webots\first_project'

>> getenv('WEBOTS_HOME')
ans =

    'C:\Program Files\Webots'

>> getenv('WEBOTS_CONTROLLER_NAME')
ans =

    'my_first_robot_controller_matlab'

>> getenv('WEBOTS_VERSION')
ans =

    'R2022a'

>> cd(getenv('WEBOTS_HOME'))
>> cd('lib/controller/matlab')
>> pwd
ans =

    'C:\Program Files\Webots\lib\controller\matlab'

>> launcher
Using prototype file: C:\Users\*****\AppData\Local\Temp\6\protofile_matlab_2022a_webots_R2022a.m
Error using loadlibrary
There was an error loading the library "Controller"
The specified module could not be found.

Error in launcher (line 138)
    loadlibrary(libname,protofile,'alias','libController');

Caused by:
    Error using loaddefinedlibrary
    The specified module could not be found.


  
**Does the variable "libname" need to be spelled with a lower case "c" rather than "Controller" as it is defined in line 35?**

**Reply by omichel** [reply]
No, that's correct. You simply have to modify also the **PATH** environment variable to add the following paths: `C:\Program Files\Webots\lib\controller` and `C:\Program Files\Webots\msys64\mingw64\bin`, as documented [here](https://cyberbotics.com/doc/guide/running-extern-robot-controllers#environment-variables): 
```MATLAB
setenv('PATH', [getenv('PATH') ';C:\Program Files\Webots\lib\controller;C:\Program Files\Webots\msys64\mingw64\bin']);
```
